smt.go

```diff
@@ -138,9 +138,10 @@ func (smt *SparseMerkleTree) HasForRoot(key, root []byte) (bool, error) {
 // Update sets a new value for a key in the tree, and sets and returns the new root of the tree.
 func (smt *SparseMerkleTree) Update(key []byte, value []byte) ([]byte, error) {
 	newRoot, err := smt.UpdateForRoot(key, value, smt.Root())
-	if err == nil {
-		smt.SetRoot(newRoot)
+	if err != nil {
+		return nil, err
 	}
+	smt.SetRoot(newRoot)
 	return newRoot, err
```

Comment:
```suggestion
	return newRoot, nil
```
To be completely idiomatic ;)